### PR TITLE
[libwebp] Update to 1.3.0, misc changes

### PR DESCRIPTION
**Pull request by dg0yt**
Misc changes:
- Add explicit usage file. (With a fifth exported target, the heuristics started to omit one.)
- Minimize patches.
- Fix tool and feature configuration (tested locally: linux, osx, mingw with `core`, default and `all`).
- Consistently depend on libjpeg-turbo, libpng, tiff for all features that could pick up these modules.
  These dependencies are used only by (some) tools, and they already existed for some features.
  There is one caveat: `tiff[webp]` cannot be installed together with `libwebp[some tools]`.
- Disable debug build of the tools.
  Not installed anyways. (Some tools are not installed even for release config...)

 _

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libwebp/vcpkg.json [hunk 1]
```diff
@@ -50,53 +58,58 @@
         {
           "name": "libwebp",
           "features": [
-            "extras"
+            "vwebp"
           ],
-          "platform": "!osx"
+          "platform": "!osx & !static"
```
Comment (Neumann-A) [inline ports/libwebp/vcpkg.json:63]: why did you add all the `!static`?

Comment (dg0yt) [reply ports/libwebp/vcpkg.json:63]: Upstream. The `vwebp` tools aren't available in static builds:
https://github.com/webmproject/libwebp/blob/b54d21a01d975c9a5681e8d3db6e0f55265ea5bb/CMakeLists.txt#L55-L65
Unless we generally disable `WEBP_LINK_STATIC` which seems to have no positive effect in vcpkg context anyways.

**Comment by dg0yt** [comment]
opencv:x64-windows: Fails to find webpmux

mapnik:x64-windows: Could NOT find WebP (missing: WebP_LIBRARY) (found version "1.3.0")
in vendored cmake/FindWebP.cmake

File: ports/libwebp/vcpkg.json (hunk 1 above)
Comment (Neumann-A) [reply ports/libwebp/vcpkg.json:63]: Hmm in that case I am less interested in what upstream says (because obviously they don't have a clue) but what the actual error is.

Comment (dg0yt) [reply ports/libwebp/vcpkg.json:63]: Me too.

**Comment by dg0yt** [comment]
Upstream changed the output name of the libraries on windows. They added the `lib` prefix:
~~~
-libwebp:x64-windows:/bin/webp.dll
-libwebp:x64-windows:/bin/webp.pdb
-libwebp:x64-windows:/bin/webpdecoder.dll
-libwebp:x64-windows:/bin/webpdecoder.pdb
-libwebp:x64-windows:/bin/webpdemux.dll
-libwebp:x64-windows:/bin/webpdemux.pdb
-libwebp:x64-windows:/bin/webpmux.dll
-libwebp:x64-windows:/bin/webpmux.pdb
-libwebp:x64-windows:/lib/webp.lib
-libwebp:x64-windows:/lib/webpdecoder.lib
-libwebp:x64-windows:/lib/webpdemux.lib
-libwebp:x64-windows:/lib/webpmux.lib
+libwebp:x64-windows:/bin/libsharpyuv.dll
+libwebp:x64-windows:/bin/libsharpyuv.pdb
+libwebp:x64-windows:/bin/libwebp.dll
+libwebp:x64-windows:/bin/libwebp.pdb
+libwebp:x64-windows:/bin/libwebpdecoder.dll
+libwebp:x64-windows:/bin/libwebpdecoder.pdb
+libwebp:x64-windows:/bin/libwebpdemux.dll
+libwebp:x64-windows:/bin/libwebpdemux.pdb
+libwebp:x64-windows:/bin/libwebpmux.dll
+libwebp:x64-windows:/bin/libwebpmux.pdb
+libwebp:x64-windows:/lib/libsharpyuv.lib
+libwebp:x64-windows:/lib/libwebp.lib
+libwebp:x64-windows:/lib/libwebpdecoder.lib
+libwebp:x64-windows:/lib/libwebpdemux.lib
+libwebp:x64-windows:/lib/libwebpmux.lib
~~~

File: ports/qt5-imageformats/portfile.cmake [hunk 2]
```diff
@@ -28,27 +28,13 @@ if(FREEGLUT_NEEDED)
...
-find_library(WEBPDECODER_RELEASE NAMES webpdecoder PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH) 
-find_library(WEBPDECODER_DEBUG NAMES webpdecoder PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
-# Depends on opengl in default build but might depend on giflib, libjpeg-turbo, zlib, libpng, tiff, freeglut (!osx), sdl1 (windows) 
-# which would require extra libraries to be linked e.g. giflib freeglut sdl1 other ones are already linked
-
-if(NOT VCPKG_TARGET_IS_WINDOWS)
-    string(APPEND WEBP_RELEASE " -pthread")
-    string(APPEND WEBP_DEBUG " -pthread")
-endif()
+x_vcpkg_pkgconfig_get_modules(PREFIX webp MODULES libwebp libwebpdemux libwebpmux libwebpdecoder LIBS)
```
Comment (Neumann-A) [inline ports/qt5-imageformats/portfile.cmake:31]: you forgot the dependency for this. (does it make sense to switch all? I mean you are already touching it.)

File: ports/qt5-imageformats/portfile.cmake [hunk 3]
```diff
@@ -28,27 +28,13 @@ if(FREEGLUT_NEEDED)
...
 
 set(OPT_REL "TIFF_LIBS=${tiff_LIBS_RELEASE}"
-            "WEBP_LIBS=${WEBPDECODER_RELEASE} ${WEBPDEMUX_RELEASE} ${WEBPMUX_RELEASE} ${WEBP_RELEASE}" 
+            "WEBP_LIBS=${webp_LIBS_RELEASE}"
             "JASPER_LIBS=${JASPER_RELEASE} ${JPEG_RELEASE} ${ZLIB_RELEASE}") # This will still fail if LIBWEBP is installed with all available features due to the missing additional dependencies
```
Comment (Neumann-A) [inline ports/qt5-imageformats/portfile.cmake:35]: is this comment still correct if pkg-config lookup is used ?

File: ports/qt5-imageformats/portfile.cmake (hunk 2 above)
Comment (dg0yt) [reply ports/qt5-imageformats/portfile.cmake:31]: There is already libwebp and there is already vcpkg-pkgconfig-get-modules. What else is missing?
The switch is done because pkgconfig properly carries the libsharpyuv dependency (now non-optional in libwebp).

File: ports/qt5-imageformats/portfile.cmake (hunk 3 above)
Comment (dg0yt) [reply ports/qt5-imageformats/portfile.cmake:35]: I didn't touch jasper. I wanted to limit the scope of this PR to libwebp. The output name change and the new internal dependency are heavy enough.

File: ports/qt5-imageformats/portfile.cmake (hunk 2 above)
Comment (Neumann-A) [reply ports/qt5-imageformats/portfile.cmake:31]: a okay i didn't see the other one in the diff. Maybe move it up so that the two `x_vcpkg_pkgconfig_get_modules` are together and not a single one in-between `find_library` stuff. Makes it easier to understand what is happening.

File: ports/libwebp/vcpkg.json (hunk 1 above)
Comment (dg0yt) [reply ports/libwebp/vcpkg.json]: Neither port freeglut nor port sdl1 provide their complete static link libraries for x64-linux. As usual, this isn't caught in CI because vcpkg doesn't build executables.
The lack of vwebp[_sdl] on static triplets isn't a regression, so I will probably not fix it at this point.
I should revert the manifest changes, so that other people can look into this with `--allow-unsupported`. i.e. this a suitable test port to fix those ports. If someone cares about sdl1.

Comment (dg0yt) [reply ports/libwebp/vcpkg.json]: ... and CMake reordering libraries due to mix of targets and filepaths.

**Review by BillyONeal — DISMISSED** [review]

**Comment by BillyONeal** [comment]
Feature testing:

```
libwebp[anim,cwebp,dwebp,extras,gif2webp,img2webp,info,libwebpmux,mux,nearlossless,simd,all,swap16bitcsp,unicode,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core]:x64-windows pass
libwebp[core,anim,cwebp,dwebp,extras,gif2webp,img2webp,info,libwebpmux,mux,nearlossless,simd,unicode,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core,cwebp,dwebp,extras,gif2webp,img2webp,info,libwebpmux,mux,nearlossless,simd,unicode,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core,anim,dwebp,extras,gif2webp,img2webp,info,libwebpmux,mux,nearlossless,simd,unicode,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core,anim,cwebp,extras,gif2webp,img2webp,info,libwebpmux,mux,nearlossless,simd,unicode,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core,anim,cwebp,dwebp,gif2webp,img2webp,info,libwebpmux,mux,nearlossless,simd,unicode,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core,anim,cwebp,dwebp,extras,img2webp,info,libwebpmux,mux,nearlossless,simd,unicode,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core,anim,cwebp,dwebp,extras,gif2webp,info,libwebpmux,mux,nearlossless,simd,unicode,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core,anim,cwebp,dwebp,extras,gif2webp,img2webp,libwebpmux,mux,nearlossless,simd,unicode,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core,anim,cwebp,dwebp,extras,gif2webp,img2webp,info,mux,nearlossless,simd,unicode,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core,anim,cwebp,dwebp,extras,gif2webp,img2webp,info,libwebpmux,nearlossless,simd,unicode,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core,anim,cwebp,dwebp,extras,gif2webp,img2webp,info,libwebpmux,mux,simd,unicode,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core,anim,cwebp,dwebp,extras,gif2webp,img2webp,info,libwebpmux,mux,nearlossless,unicode,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core,anim,cwebp,dwebp,extras,gif2webp,img2webp,info,libwebpmux,mux,nearlossless,simd,vwebp,vwebp-sdl]:x64-windows pass
libwebp[core,anim,cwebp,dwebp,extras,gif2webp,img2webp,info,libwebpmux,mux,nearlossless,simd,unicode,vwebp-sdl]:x64-windows pass
libwebp[core,anim,cwebp,dwebp,extras,gif2webp,img2webp,info,libwebpmux,mux,nearlossless,simd,unicode,vwebp]:x64-windows pass
```

**Comment by BillyONeal** [comment]
Sorry I'm fixing the patch now...

**Comment by BillyONeal** [comment]
It looks like the patch added to mapnik is no longer necessary so I'm just reverting those changes.

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
@dg0yt Please confirm that the merge I did looks OK to you :)

**Comment by dg0yt** [comment]
Okay.
Mapnik switch from a vendored CMake Find module to pkg-config (via CMake's FindPkgConfig.cmake). This works as good or as bad as anything else.

**Comment by BillyONeal** [comment]
Thanks!
